src/components/search/filter-group.js

```diff
@@ -0,0 +1,158 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import { injectIntl, FormattedMessage } from 'react-intl'
+import $ from 'jquery'
+
+import MultipleSelectionFilter from 'components/search/multiple-selection-filter'
+import PriceFilter from 'components/search/price-filter'
+import CounterFilter from 'components/search/counter-filter'
+import DateFilter from 'components/search/date-filter'
+import { updateFilters } from 'actions/Search'
+
+class FilterGroup extends Component {
+  constructor(props) {
+    super(props)
+
+    this.childFilters = []
+
+    this.title = this.props.intl.formatMessage(this.props.filterGroup.title)
+    this.handleFilterMounted = this.handleFilterMounted.bind(this)
+    this.handleFilterUnMounted = this.handleFilterUnMounted.bind(this)
+    this.handleApplyClick = this.handleApplyClick.bind(this)
+    this.handleClearClick = this.handleClearClick.bind(this)
+    this.handleOpenDropdown = this.handleOpenDropdown.bind(this)
+  }
+
+  resolveFromListingSchema(path) {
+    var properties = Array.isArray(path) ? path : path.split('.')
+    return properties.reduce((prev, curr) => prev && prev[curr], this.props.listingSchema)
+  }
+
+  handleFilterMounted(filter) {
+    this.childFilters.push(filter)
+  }
+
+  handleFilterUnMounted(filter) {
+    const index = this.childFilters.indexOf(filter)
+    if (index !== -1)
+      this.childFilters.splice(index, 1)
+  }
+
+  handleApplyClick(event) {
+    event.preventDefault()
+
+    const filters = this.childFilters
+      .flatMap(childFilter => childFilter.getFilters())
+
+    this.props.updateFilters(this.title, filters)
+
+    // close the dropdown menu. Handles the css clases and aria-expanded attribute
+    $("body").trigger("click")
+  }
+
+  handleClearClick(event) {
+    event.preventDefault()
+
+    this.childFilters
+      .forEach(childFilter => childFilter.onClear())
+  }
+
+  handleOpenDropdown(event) {
+    const containsDateFilter = this.props.filterGroup.items.some(filter => filter.type == 'date')
+    if (!containsDateFilter)
+      return
+
+    /* Because of a workaround in `react-dates` module we need to message the DateFilter
+     * when it gets shown - when dropdown containing date filter is opened
+     */
+    this.childFilters
+      .filter(filter => filter.props.filter.type == 'date')
+      .forEach(dateFilter => dateFilter.onOpen())
+  }
+
+  renderFilter(filter, title) {
+    if (filter.type == 'multipleSelectionFilter') {
+      return(
+        <MultipleSelectionFilter
+          filter={filter}
+          multipleSelectionValues={this.resolveFromListingSchema(filter.listingPropertyName)}
+          listingType={this.props.listingType}
+          title={title}
+          onChildMounted={this.handleFilterMounted}
+          onChildUnMounted={this.handleFilterUnMounted}
+        />
+      )
+    } else if (filter.type == 'price') {
```

Comment:
Let's use strict [in]equality everywhere unless there is a compelling reason not to. 🔒